docs/rules-engine/examples/portal-facts.md

```diff
@@ -0,0 +1,123 @@
+# Rules Engine Example - Portal Facts
+
+## Pre-requisite
+
+This use case applies to applications that are targeted via deeplink (for example, from an email or an external 
+application portal).
+
+## Introduction
+
+For this example, we want to reference the data of the deeplink in the Rulesets of the application. For this purpose, the
+data will be transferred to the application in the deeplink ``portalFacts`` parameter.
+
+This feature should be generic. Neither the metadata nor the code of the application should need to be updated to
+support new parameters in order to simplify the integration with external applications.
+
+Let's check how to meet these requirements.
+
+## Implementation
+
+### PortalFacts
+
+The deeplink parameters are exposed to the rule engine via a ``portalFacts`` fact which will emit a map of `<string, string>` with
+the different portal parameters name associated to their values.
+
+> [!WARNING]
+>
+> At the moment, `portalFacts` only supports string and no complex value.
+
+The rule will use the ``portalFacts`` with a json path to select the requested parameter.
+
+Note that, as its content is dynamic, ``portalFacts`` should not be part of the [metadata of the rules engine](../how-to-use/industrialize-ruleset-generation.md).
+To avoid its extraction, it will be part of a dedicated folder. 
+
+You will need to generate a ``PortalFactService`` in this folder and register it in the application rules engine.
+For this purpose, please follow the [custom fact documentation](../how-to-use/custom-fact.md).
+
+This service will expose a fact with the ``portalFacts`` object sent by the deeplink.
+
+For this use case, we will take the assumption that ``portalFacts`` is sent as a POST parameter and injected in the index.html
+``data-post`` property.
+
+This means that the ``https://url-of-my-app?portalFacts=[{"key":"portalFactName1", "value":"portalFactValue2"}]&`` request will return this html :
+
+```html
+
+<body data-post='{"portalFacts":[{"key":"portalFactName1","value":"portalFactValue2"}]}'>
+    ...
+
+</body>
+```
+ 
+This way, we can use the Otter ``RequestParametersService`` ([@o3r/dynamic-content](https://www.npmjs.com/package/@o3r/dynamic-content)) 
+to retrieve the object and use it to set up the ``portalFacts`` map.
+
+```typescript
+import {Injectable} from '@angular/core';
+import {FactsService, PortalFacts, RulesEngineRunnerService, FactSet} from '@o3r/rules-engine';
+import {RequestParametersService} from '@o3r/dynamic-content';
+import {of} from 'rxjs';
+
+@Injectable()
+export class PortalFactsService extends FactsService<PortalFacts> {
+
+  public facts: FactSet<PortalFacts>;
+
+  constructor(rulesEngine: RulesEngineRunnerService, requestParametersService: RequestParametersService) {
+    super(rulesEngine);
```

Comment:
maybe we can do the example with the `inject` method to follow the new Angular guideline